test/controllers/SubmitApplicationControllerSpec.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.{ CSRCache, CSRHttp }
+import connectors.ApplicationClient
+import models.ApplicationRoute._
+import models.SecurityUserExamples._
+import models.{ CachedData, CachedDataExample, CachedDataWithApp, ProgressResponseExamples }
+import org.mockito.Matchers.{ eq => eqTo, _ }
+import org.mockito.Mockito._
+import play.api.test.Helpers._
+import security.UserService
+import testkit.BaseControllerSpec
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+
+class SubmitApplicationControllerSpec extends BaseControllerSpec {
+
+  override def currentCandidateWithApp: CachedDataWithApp = CachedDataWithApp(ActiveCandidate.user,
+    CachedDataExample.InProgressInPreviewApplication.copy(userId = ActiveCandidate.user.userID))
+
+  "present" should {
+    "display submit application page" in new TestFixture {
+      val applicationRouteConfig = ApplicationRouteConfig(newAccountsStarted = true,
+        newAccountsEnabled = true, applicationsSubmitEnabled = true)
+      val result = controller(currentCandidateWithEdipApp, applicationRouteConfig).present()(fakeRequest)
+      status(result) mustBe OK
+      val content = contentAsString(result)
+      content must include("Submit application")
+    }
+    "redirect to home page" in new TestFixture {
```

Comment:
there could be a little more description in the test name to describe the conditions you are expecting a submit/redirect